### PR TITLE
BUG 1974816: util: allow any KMS connection to be configured by ConfigMap

### PR DESCRIPTION
**Pull request by nixpanic**
The ConfigMap that is created by the OCS UI contains a very specific set
of options. Ceph-CSI does have an implementation that supports Hashicorp
Vault with a ServiceAccount, but this can not be configured through the
UI.

By allowing manual changes to the ConfigMap, and keeping the options in
the configuration, users will be able to configure a Hashicorp Vault
connection that uses a ServiceAccount to connect to the KMS.

- [x] **WARNING** downstream only change!
- [x] is manually tested
- [x] contains unit tests
- [ ] verified non-breaking from release-4.8

### CODE REVIEW

**Comment by agarwal-mudit** [comment]
cc @Rakshith-R

File: internal/util/vault_tokens.go [hunk 1]
```diff
@@ -110,23 +180,27 @@ func getVaultConfiguration(namespace, name string) (map[string]interface{}, erro
...
+		if ch.isVaultTokensConfigMap() {
+			sv, err := dataToStandardVault(k, v)
+			if err != nil {
+				return nil, fmt.Errorf("could not parse Vault config for %q: %w", k, err)
+			}
+			config[k] = sv
+		} else {
+			options, err := dataToMap(v)
+			if err != nil {
+				return nil, fmt.Errorf("coult not parse options for %q: %w", k, err)
```
Comment (humblec) [inline internal/util/vault_tokens.go]: s/coult/could

Comment (nixpanic) [reply internal/util/vault_tokens.go]: updated!

**Comment by humblec** [comment]
Thanks @nixpanic , LGTM. I am not sure about  disabling the `iss` part in general deployment side (at customer's/user's end) though.

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Comment by nixpanic** [comment]
> Thanks @nixpanic , LGTM. I am not sure about disabling the `iss` part in general deployment side (at customer's/user's end) though.

That is just for the test deployment of Hashicorp Vault. I had to disable iss validation because recent OCP versions are based on Kubernetes v1.21. See external-secrets/kubernetes-external-secrets#721 for more details about the change in Kubernetes.

**Pull request merged**